### PR TITLE
fix: mypy types + pylint

### PR DESCRIPTION
**Pull request by himanshu-dixit**
- FIx all the typing issues
- Add generic to avoid Liskov substition principle https://stackoverflow.com/questions/54346721/mypy-argument-of-method-incompatible-with-supertype
- ignore type stub error

### CODE REVIEW

**Comment by sohamganatra** [comment]
Can we correct linting checks

File: composio/cli/__init__.py [hunk 1]
```diff
@@ -14,9 +12,17 @@
...
+from composio.core.cls.catch_all_exceptions import (
+    CatchAllExceptions,
+    handle_exceptions,
+    init_sentry,
+)
 from composio.core.cls.did_you_mean import DYMGroup
-from composio.cli.utils import HelpfulCmdBase
+
+
+init_sentry()
```
Comment (angrybayblade) [inline composio/cli/__init__.py:25]: Can we perform lazy loading here? I think this will slow down the CLI

File: .pylintrc [hunk 2]
```diff
@@ -3,7 +3,7 @@ ignore-patterns=
 ignore=
 
 [MESSAGES CONTROL]
-disable=C0114,C0115,C0116,C0301,C0103,W0603,R1715,W0621,R0903,W0237,W0511,W0622,R0913,R0902,W0221,C0302,R0801,C0411,C0412,W0719,W0718,R0914,R0916,R0912,R0911,W0102
+disable=C0114,C0115,C0116,C0301,C0103,W0603,R1715,W0621,R0903,W0237,W0511,W0622,R0913,R0902,W0221,C0302,R0801,C0411,C0412,W0719,W0718,R0914,R0916,R0912,R0911,W0102,C0325
```
Comment (angrybayblade) [inline .pylintrc:6]: Please add a comment about the disabled message

File: composio/cli/utils/helpfulcmd.py [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
-    help = None
+    examples: list[str] = []
+    help: t.Optional[str] = None
 
-    def format_help_text(self, ctx: ClickContext, formatter: HelpFormatter) -> None:
+    def format_help_text(
+        self,
+        ctx: ClickContext,  # pylint: disable=unused-argument
+        formatter: HelpFormatter,  # pylint: disable=unused-argument
+    ) -> None:
```
Comment (angrybayblade) [inline composio/cli/utils/helpfulcmd.py:17]:
```suggestion
    def format_help_text(  # pylint: disable=unused-argument
        self,
        ctx: ClickContext,
        formatter: HelpFormatter, 
    ) -> None:
```

File: composio/cli/apps.py [hunk 4]
```diff
@@ -275,8 +275,8 @@ def _get_action_enum(apps: t.List[AppModel], actions: t.List[ActionModel]) -> st
     local_tool_handler = LocalToolHandler()
     for tool in local_tool_handler.registered_tools:
         for action in tool.actions():
-            enum_name = f"{_get_enum_key(action().get_tool_merged_action_name())}"
-            enum_value = f'("{tool.tool_name}", "{tool.tool_name}_{action().action_name}", True, True)'
+            enum_name = f"{_get_enum_key(action().get_tool_merged_action_name())}"  # type: ignore
+            enum_value = f'("{tool.tool_name}", "{tool.tool_name}_{action().action_name}", True, True)'  # type: ignore
```
Comment (kaavee315) [inline composio/cli/apps.py:279]: What is the exact issue here?

File: composio/cli/utils/helpfulcmd.py [hunk 5]
```diff
@@ -33,7 +33,7 @@ def format_help_text(self, ctx: ClickContext, formatter: HelpFormatter) -> None:
     def format_options(self, ctx: ClickContext, formatter: HelpFormatter) -> None:
         """Writes all the options into the formatter if they exist."""
         opts = []
-        for param in self.get_params(ctx):
+        for param in self.get_params(ctx):  # type: ignore
```
Comment (kaavee315) [inline composio/cli/utils/helpfulcmd.py:36]: Hmmm seems a simple line, what's the issue.

File: composio/client/__init__.py [hunk 6]
```diff
@@ -716,17 +716,16 @@ def get(  # type: ignore
...
+            should_not_filter_using_tags = (len(items) < 15 and only_important_tag)
             if not should_not_filter_using_tags:
-                items = [
+                filtered_items = [
                     item
                     for item in items
                     if any(tag in required_tags for tag in item.tags)
                 ]
+                if len(filtered_items) > 0 or not only_important_tag:
+                    items = filtered_items
```
Comment (kaavee315) [inline composio/client/__init__.py:728]: Mind rebasing?

**Review by kaavee315 — APPROVED** [review]

File: composio/cli/apps.py (hunk 4 above)
Comment (shubhras01) [reply composio/cli/apps.py:279]: cc - @himanshu-dixit

File: composio/cli/utils/helpfulcmd.py (hunk 5 above)
Comment (shubhras01) [reply composio/cli/utils/helpfulcmd.py:36]: cc - @himanshu-dixit

File: composio/client/__init__.py [hunk 7]
```diff
@@ -1,15 +1,17 @@
+# fmt: off
```
Comment (angrybayblade) [inline composio/client/__init__.py:1]: Why are we turning off code formatting here?

File: composio/client/__init__.py [hunk 8]
```diff
@@ -765,14 +764,18 @@ def execute(
...
-                with open(value, 'rb') as file:
+                with open(value, "rb") as file:
                     file_content = file.read()
                     try:
-                        file_content.decode('utf-8')  # Try decoding as UTF-8 to check if it's normal text
-                        modified_params[param] = file_content.decode('utf-8')
+                        file_content.decode(
+                            "utf-8"
+                        )  # Try decoding as UTF-8 to check if it's normal text
+                        modified_params[param] = file_content.decode("utf-8")
```
Comment (angrybayblade) [inline composio/client/__init__.py:773]:
```suggestion
                        modified_params[param] = file_content.decode("utf-8")
```

File: composio/core/local/action.py [hunk 9]
```diff
@@ -85,56 +91,77 @@ def get_tool_merged_action_name(self) -> str:
...
-                    with open(value, 'rb') as file:
+                    with open(value, "rb") as file:
                         file_content = file.read()
                         try:
-                            file_content.decode('utf-8')  # Try decoding as UTF-8 to check if it's normal text
-                            modified_request_data[param] = file_content.decode('utf-8')
+                            file_content.decode(
+                                "utf-8"
+                            )  # Try decoding as UTF-8 to check if it's normal text
+                            modified_request_data[param] = file_content.decode("utf-8")
```
Comment (angrybayblade) [inline composio/core/local/action.py:153]:
```suggestion
                            modified_request_data[param] = file_content.decode("utf-8")
```

File: composio/local_tools/ragtool/actions/rag_query.py [hunk 10]
```diff
@@ -25,7 +25,7 @@ class RagToolQuery(Action):
     _tags = ["Knowledge Base"]
     _tool_name = "ragtool"
 
-    def execute(self, request: RagToolQueryRequest, authorisation_data: dict = None):
+    def execute(self, request: RagToolQueryRequest, authorisation_data: dict) -> dict:
```
Comment (angrybayblade) [inline composio/local_tools/ragtool/actions/rag_query.py:28]:
```suggestion
    def execute(self, request_data: RagToolQueryRequest, authorisation_data: dict) -> dict:
```

https://github.com/ComposioHQ/composio/blob/42debbddf3cb2f0e7d77f3f6eaff2f9ea9495a2e/composio/core/local/action.py#L81

File: composio/local_tools/ragtool/actions/rag_add_request.py [hunk 11]
```diff
@@ -4,25 +4,29 @@
...
     _display_name = "Add Content to Rag Tool"
-    _request_schema = RagToolAddRequest
-    _response_schema = RagToolAddResponse
+    _request_schema: type[RagToolAddRequest] = RagToolAddRequest
+    _response_schema: type[RagToolAddResponse] = RagToolAddResponse
     _tags = ["Knowledge Base"]
     _tool_name = "ragtool"
 
-    def execute(self, request: RagToolAddRequest, authorisation_data: dict = None):
+    def execute(self, request: RagToolAddRequest, authorisation_data: dict) -> dict:
```
Comment (angrybayblade) [inline composio/local_tools/ragtool/actions/rag_add_request.py:29]:
```suggestion
    def execute(self, request_data: RagToolAddRequest, authorisation_data: dict) -> dict:
```

https://github.com/ComposioHQ/composio/blob/42debbddf3cb2f0e7d77f3f6eaff2f9ea9495a2e/composio/core/local/action.py#L81

File: composio/local_tools/greptile/actions/codequery.py [hunk 12]
```diff
@@ -65,7 +65,7 @@ class CodeQuery(Action):
     _tool_name = "greptile"
 
     def execute(
-        self, request_data: CodeQueryRequest, authorisation_data: dict = {}
+        self, request_data: CodeQueryRequest, authorisation_data: dict = {}  # type: ignore[override]
```
Comment (angrybayblade) [inline composio/local_tools/greptile/actions/codequery.py:68]:
```suggestion
        self, request_data: CodeQueryRequest, authorisation_data: dict  # type: ignore[override]
```

File: composio/local_tools/file/actions/read_file.py [hunk 13]
```diff
@@ -27,17 +28,19 @@ class ReadFileResponse(BaseModel):
...
-    _response = ReadFileResponse
+    _request: Type[ReadFileRequest] = ReadFileRequest
+    _response: Type[ReadFileResponse] = ReadFileResponse
     _tags = ["file", "read"]
     _tool_name = "file"
 
-    def execute(self, request: ReadFileRequest) -> ReadFileResponse:
+    def execute(
+        self, request: ReadFileRequest, authorisation_data: dict
+    ) -> ReadFileResponse:
```
Comment (angrybayblade) [inline composio/local_tools/file/actions/read_file.py:43]:
```suggestion
    def execute(
        self, request_data: ReadFileRequest, authorisation_data: dict
    ) -> ReadFileResponse:
```

**Pull request merged**